metadata-ingestion/src/datahub/ingestion/source/airbyte/rest_api_wrapper/data_resolver.py

```diff
@@ -0,0 +1,468 @@
+                namespace_format=response_dict[Constant.NAMESPACEFORMAT],
+                prefix=response_dict[Constant.PREFIX],
+                jobs=self._get_jobs(response_dict[Constant.CONNECTIONID]),
+            )
+            for response_dict in response
+        ]
+        return connections
+
+    def _get_server_from_configuration(
+        self, configuration: Dict, connector_type: str
+    ) -> str:
+        return configuration[self.CONNECTOR_SERVER_KEY_MAPPING[connector_type]]
+
+    def _get_source_endpoint(self, source_id: str) -> str:
+        source_endpoint: str = self.API_ENDPOINTS[Constant.SOURCE_GET]
+        # Replace place holders
+        return source_endpoint.format(API_URL=self.api_url, SOURCE_ID=source_id)
+
+    def _get_source_connector_from_response(self, response: Dict) -> Connector:
+        connector_type = (
+            response[Constant.SOURCETYPE]
+            if Constant.SOURCETYPE in response
+            else response[Constant.SOURCENAME]
+        ).lower()
+        configuration = (
+            response[Constant.CONNECTIONCONFIGURATION]
+            if Constant.CONNECTIONCONFIGURATION in response
+            else response[Constant.CONFIGURATION]
+        )
+        return Connector(
+            connector_id=response[Constant.SOURCEID],
+            name=response[Constant.NAME],
+            type=connector_type,
+            server=self._get_server_from_configuration(configuration, connector_type),
+        )
+
+    def _get_destination_endpoint(self, destination_id: str) -> str:
+        destination_endpoint: str = self.API_ENDPOINTS[Constant.DESTINATION_GET]
+        # Replace place holders
+        return destination_endpoint.format(
+            API_URL=self.api_url, DESTINATION_ID=destination_id
+        )
+
+    def _get_destination_connector_from_response(self, response: Dict) -> Connector:
+        connector_type = (
+            response[Constant.DESTINATIONTYPE]
+            if Constant.DESTINATIONTYPE in response
+            else response[Constant.DESTINATIONNAME]
+        ).lower()
+        configuration = (
+            response[Constant.CONNECTIONCONFIGURATION]
+            if Constant.CONNECTIONCONFIGURATION in response
+            else response[Constant.CONFIGURATION]
+        )
+        return Connector(
+            connector_id=response[Constant.DESTINATIONID],
+            name=response[Constant.NAME],
+            type=connector_type,
+            server=self._get_server_from_configuration(configuration, connector_type),
+        )
+
+    def _get_jobs_endpoint(self, connection_id: str) -> str:
+        jobs_endpoint: str = self.API_ENDPOINTS[Constant.JOBS_LIST]
+        # Replace place holders
+        return jobs_endpoint.format(API_URL=self.api_url, CONNECTION_ID=connection_id)
+
+    def _get_jobs_from_response(self, response: List[Dict]) -> List[Job]:
+        jobs: List[Job] = [
+            Job(
```

Comment:
what is the purpose of these if else? lets discuss this in call

Reply:
Aibyte cloud and OSS response contains some same metadata with different key.
Instead of defining seperate function in both CloudApiResolver and OssApiResolver I did this and define one function in abstraction class.